src/tools/tcc.lua

```diff
@@ -0,0 +1,500 @@
+---
+-- tcc.lua
+-- Provides Tiny C Compiler-specific configuration strings.
+-- Copyright (c) 2002-2023  Perkins and the Premake project
+---
+
+	local p = premake
+
+	p.tools.tcc = {}
+	local tcc = p.tools.tcc
+
+	local project = p.project
+	local config = p.config
+
+
+--
+-- Returns list of C preprocessor flags for a configuration.
+--
+
+	tcc.cppflags = {
```

Comment:
Most stuff seems to be a copy of (old) gcc functions/tables.
if true, maybe do as clang and just do `tcc.XXX = gcc.XXX`
I said *old* as some stuff seems missing (standard "C17", openmp, ...)

Reply:
It is indeed, and while I agree that basing off of GCC could work due to similarities, it would strip out a lot more so i figured it was better to maintain a separate file. Is this okay?

Reply:
If it is different, then separate fields are ok.